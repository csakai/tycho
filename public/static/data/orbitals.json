[{
	"name": "Earth",
	"albedo": 0.367,
	"sidereal": 0.99726968,
	"radius": 6371,
	"markdown": "earth.md",
	"mass": 5.97237e+24,
	"centralMass": 1.989e+30,
	"axialTilt": 23.4392811,
	"inclination": 7.155,
	"atmosphere": 35260,
	"satellites": [{
		"name": "Moon",
		"albedo": 0.136,
		"sidereal": 0.99726968,
		"radius": 1737.1,
		"markdown": "moon.md",
		"mass": 5.97237e+24,
		"centralMass": 5.97237e+24,
		"axialTilt": 6.687,
		"inclination": 5.145,
		"atmosphere": 10066278,
		"ephemeris": {
			"nasaOrbitalCode": "301",
			"nasaBarycenterCode": "3",
			"revolutionOrder": "d"
		},
		"argumentOfPeriapsis": 282.5992409518417,
		"eccentricity": 0.03405532141489297,
		"longitudeOfAscendingNode": 142.3595722910625,
		"semimajor": 376126.5103019063,
		"semiminor": 375908.33786092995,
		"periapses": {
			"last": 1509927058416.626,
			"next": 1512377208118.1555
		},
		"id": "moon"
	}],
	"ephemeris": {
		"nasaOrbitalCode": "3",
		"nasaBarycenterCode": "0",
		"revolutionOrder": "y"
	},
	"argumentOfPeriapsis": 114.20763,
	"eccentricity": 0.01355360061112275,
	"longitudeOfAscendingNode": 348.73936,
	"semimajor": 150759239.9570005,
	"semiminor": 150745392.07806867,
	"periapses": {
		"last": 1136430000000,
		"next": 1167987600000
	},
	"id": "earth"
},{
	"name": "Venus",
	"albedo": 0.689,
	"sidereal": -243.025,
	"radius": 6051.8,
	"markdown": "venus.md",
	"mass": 4.8675e+24,
	"centralMass": 1.989e+30,
	"axialTilt": 2.64,
	"inclination": 3.86,
	"atmosphere": 16052641,
	"ephemeris": {
		"nasaOrbitalCode": "2",
		"nasaBarycenterCode": "0",
		"revolutionOrder": "y"
	},
	"argumentOfPeriapsis": 55.186,
	"eccentricity": 0.0067,
	"longitudeOfAscendingNode": 76.678,
	"semimajor": 108208179,
	"semiminor": 107477094,
	"periapses": {
		"last": 1429348560000,
		"next": 1448787300000
	},
	"id": "venus"
}]