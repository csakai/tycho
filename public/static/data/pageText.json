{"planetScale":"Planet Scale: × ","speedScale":"Speed Scale: × 10^","settings":"Settings"}
