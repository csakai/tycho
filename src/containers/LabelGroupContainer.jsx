import React from 'react';
import PropTypes from 'prop-types';
import LabelContainer from './LabelContainer';
import {connect} from 'react-redux';
import ReduxService from '../services/ReduxService';

export class LabelGroupContainer extends React.Component {

  static propTypes = {
    orbitalData: PropTypes.array.isRequired
  }
  
  getOrbitalLabels = (orbitals) => {
    return orbitals.map((orbital) => (
      <LabelContainer
        position={this.props.positions && this.props.positions[orbital.id]}
        text={orbital.name}
        id={orbital.id}
        key={orbital.id}>
        {orbital.satellites && this.getOrbitalLabels(orbital.satellites)}
      </LabelContainer>
    ));
  }

  render() {
    return <div>{this.getOrbitalLabels(this.props.orbitalData)}</div>;
  }
}

export default connect(
  ReduxService.mapStateToProps(
    'uiControls.positions'
  ),
  null
)(LabelGroupContainer);
