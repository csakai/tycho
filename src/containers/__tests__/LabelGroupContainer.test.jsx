import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {LabelGroupContainer} from '../LabelGroupContainer';
import data from '../../global/fixtures';

describe('Label Group Container', () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <LabelGroupContainer
        orbitalData={data}
        positions={{}}
      />);
  });
  
  it('should render the LabelGroup successfully', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
