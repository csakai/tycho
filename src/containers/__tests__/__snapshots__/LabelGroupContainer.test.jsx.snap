// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Group Container should render the LabelGroup successfully 1`] = `
<div>
  <Connect(LabelContainer)
    id="dummyOuter"
    key="dummyOuter"
    text="Dummy Outer"
  />
  <Connect(LabelContainer)
    id="dummyParent"
    key="dummyParent"
    text="Dummy Planet"
  >
    <Connect(LabelContainer)
      id="dummyChild"
      key="dummyChild"
      text="Dummy Satellite"
    />
  </Connect(LabelContainer)>
</div>
`;
